### DISCUSSION TITLE
Is auto-import supposed to work with 3rd party packages?

### DISCUSSION THREAD

**Discussion by rodrigoscc** [python-lsp/python-lsp-server#355] [General]
I have rope auto-import enabled and it works for built-in functions and classes. For instance, if I type `ip_ad`, the `ipaddress.ip_address` function will be listed in the completion popup, and if I pick it, it will be automatically imported. However, if I type `MethodView`, I don't see `flask.views.MethodView` listed in the completion popup.

I made sure to activate the virtual environment with Flask installed before opening Neovim. So my question is, is this expected behavior or is there something wrong with my setup? Thanks.

Pylsp version: 1.7.1
Neovim: 0.8.3

**Comment by rodrigoscc** [comment]
Alright, I figured it out. I forgot to update the `python_path` rope setting.

For those interested, I added this to my project `.ropeproject/config.py` file:
```
def set_prefs(prefs):
    prefs.add('python_path', '/Users/rodrigo/python-project/venv/lib/python3.9/site-packages/')
```

**Comment by ccordoba12** [comment]
@rstcruzo, thanks for mentioning a solution!

@bagel897, is there something we could about this on pylsp directly?

**Comment by bagel897** [comment]
Pylsp would need to run in the same virtual environment/be installed there. Otherwise you can pass the environment as a parameter to initialize the rope project in workspace.py

**Reply by ccordoba12** [reply]
> Otherwise you can pass the environment as a parameter to initialize the rope project in workspace.py

Yeah, that's what I was thinking. Users could send a didChangeConfiguration to set their current interpreter for the Rope project we use for refactorings.

**Reply by bagel897** [reply]
I would hold off on doing the didChangeConfiguration part until [this issue](https://github.com/python-rope/rope/issues/565) and maybe [this PR](https://github.com/python-rope/rope/pull/516) are resolved for speed/consistency reasons. Bit busy rn but shouldn't be too hard for someone to implement. First, I'd implement a general environment path for pylsp, possibly with PEP 582 support.

**Reply by fabio-pardo** [reply]
Has there been any update with this topic? I'm interested ☻
